storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogSegmentMetadata.java

```diff
@@ -270,9 +247,9 @@ public String toString() {
                "remoteLogSegmentId=" + remoteLogSegmentId +
                ", startOffset=" + startOffset +
                ", endOffset=" + endOffset +
-               ", brokerId=" + brokerId +
+               ", brokerId=" + brokerId() +
```

Comment:
Perhaps it is worth considering using `StringBuilder` here given that we are doing a number of string concatenations (~10) now.

https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html

Reply:
Interesting, there are tons of references to `StringBuilder` in the AK code base today: https://github.com/apache/kafka/search?p=1&q=StringBuilder. Are you saying most of these are overkill?